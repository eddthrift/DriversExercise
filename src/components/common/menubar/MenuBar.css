.menu-container{
    background-color: red;
    padding-bottom: 25px;
}

.menu-link{
    color: white;
    font-size: x-large;
    padding-top: 25px;
    padding-left: 30px;
    text-decoration: none;
}

.menu-link:hover{
    color: lightgray;
}