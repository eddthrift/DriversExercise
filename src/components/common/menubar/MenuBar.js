import React from 'react';
import {Link} from 'react-router-dom';
import {Box, Stack} from '@mui/material';
import MenuData from '../../../data/menu.json'

import './MenuBar.css';

function MenuBar(){
    return(
        <Box className="menu-container">
            <Stack>
                {MenuData.data.map(function(item) {
                    return(
                        <Link key={item.title} to={item.url} className="menu-link">&bull;&nbsp;&nbsp;&nbsp;{item.title}</Link>
                    );
                })}
            </Stack>
        </Box>
    );
};

export default MenuBar;