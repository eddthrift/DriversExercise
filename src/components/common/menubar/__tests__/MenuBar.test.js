import React from 'react';
import renderer from 'react-test-renderer';

import MenuBar from '../MenuBar';
import { MemoryRouter } from 'react-router-dom';

it('renders correctly', () => {
    const tree = renderer.create(
    <MemoryRouter>
        <MenuBar />
    </MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
});