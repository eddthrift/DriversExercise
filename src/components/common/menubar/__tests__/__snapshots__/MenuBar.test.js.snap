// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="menu-container MuiBox-root css-0"
>
  <div
    className="MuiStack-root css-nen11g-MuiStack-root"
  >
    <a
      className="menu-link"
      href="/"
      onClick={[Function]}
    >
      •   
      Home
    </a>
    <a
      className="menu-link"
      href="/drivers"
      onClick={[Function]}
    >
      •   
      Drivers
    </a>
    <a
      className="menu-link"
      href="/vehicles"
      onClick={[Function]}
    >
      •   
      Vehicles
    </a>
    <a
      className="menu-link"
      href="/about"
      onClick={[Function]}
    >
      •   
      About
    </a>
  </div>
</div>
`;
